File: webpack/javascript/services/jackson_square_calculator.js
export default class JacksonSquareCalculator {
  price(length, width, pricePer) {
    return length * width * pricePer + (length + width)
  }

  roundedPrice(length, width, pricePer, roundingScheme) {
    let price = this.price(length, width, pricePer)

    return new roundingScheme().round(price)
  }
}

File: services/jackson_square_calculator.rb
# frozen_string_literal: true

require_relative './application_service'

# Calculates a price per square inch
class JacksonSquareCalculator < ApplicationService
  def initialize(length, width, price_per)
    @length = length
    @width = width
    @price_per = price_per
  end

  def call
    (@length * @width * @price_per) + (@length + @width)
  end
end
